modules/tfm/tfm/boards/common/crypto_keys.c

```diff
@@ -81,144 +76,71 @@ static enum tfm_plat_err_t tfm_plat_get_iak(uint8_t *buf, size_t buf_len,
 
 	return TFM_PLAT_ERR_SUCCESS;
 }
-#endif /* TFM_PARTITION_INITIAL_ATTESTATION */
 
-enum tfm_plat_err_t tfm_plat_builtin_key_get_usage(psa_key_id_t key_id,
-						   mbedtls_key_owner_id_t user,
-						   psa_key_usage_t *usage)
-{
-	*usage = 0;
 
-	switch (key_id) {
-#ifdef CONFIG_HW_UNIQUE_KEY
-	case TFM_BUILTIN_KEY_ID_HUK:
-		/* Allow access to all partitions */
-		*usage = PSA_KEY_USAGE_DERIVE;
-		break;
-#endif /* CONFIG_HW_UNIQUE_KEY*/
-
-#ifdef TFM_PARTITION_INITIAL_ATTESTATION
-	case TFM_BUILTIN_KEY_ID_IAK:
-		switch(user) {
-		case TFM_SP_INITIAL_ATTESTATION:
-			*usage = PSA_KEY_USAGE_SIGN_HASH;
+/**
+ * @brief Table describing per-user key policy for the IAK
+ *
+ */
+static const tfm_plat_builtin_key_per_user_policy_t g_iak_per_user_policy[] = {
+    {.user = TFM_SP_INITIAL_ATTESTATION,
 #ifdef SYMMETRIC_INITIAL_ATTESTATION
-		/* Needed to calculate the instance ID */
-			*usage |= PSA_KEY_USAGE_EXPORT;
+        .usage = PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_EXPORT,
+#else
+        .usage = PSA_KEY_USAGE_SIGN_HASH,
 #endif /* SYMMETRIC_INITIAL_ATTESTATION */
-			break;
-
-#if defined(TEST_S_ATTESTATION) || defined(TEST_NS_ATTESTATION)
-		/* So that the tests can validate created tokens */
+    },
 #ifdef TEST_S_ATTESTATION
-		case TFM_SP_SECURE_TEST_PARTITION:
+    {.user = TFM_SP_SECURE_TEST_PARTITION, .usage = PSA_KEY_USAGE_VERIFY_HASH},
 #endif /* TEST_S_ATTESTATION */
 #ifdef TEST_NS_ATTESTATION
-		case TFM_NS_PARTITION_ID:
+    {.user = TFM_NS_PARTITION_ID, .usage = PSA_KEY_USAGE_VERIFY_HASH},
 #endif /* TEST_NS_ATTESTATION */
-			*usage = PSA_KEY_USAGE_VERIFY_HASH;
-			break;
-#endif /* TEST_S_ATTESTATION || TEST_NS_ATTESTATION */
-
-		default:
-			return TFM_PLAT_ERR_NOT_PERMITTED;
-		}
-
-		break;
+};
 #endif /* TFM_PARTITION_INITIAL_ATTESTATION */
-	default:
-		return TFM_PLAT_ERR_UNSUPPORTED;
-	}
-
-	return TFM_PLAT_ERR_SUCCESS;
-}
-
-enum tfm_plat_err_t tfm_plat_builtin_key_get_lifetime_and_slot(mbedtls_svc_key_id_t key_id,
-							       psa_key_lifetime_t *lifetime,
-							       psa_drv_slot_number_t *slot_number)
-{
-	switch (MBEDTLS_SVC_KEY_ID_GET_KEY_ID(key_id)) {
-
-#ifdef CONFIG_HW_UNIQUE_KEY
-	case TFM_BUILTIN_KEY_ID_HUK:
-		*slot_number = TFM_BUILTIN_KEY_SLOT_HUK;
-		*lifetime = PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(
-			PSA_KEY_LIFETIME_PERSISTENT,
-			TFM_BUILTIN_KEY_LOADER_KEY_LOCATION);
-		break;
-#endif /* CONFIG_HW_UNQUE_KEY */
 
+/**
+ * @brief Table describing per-key user policies
+ *
+ */
+static const tfm_plat_builtin_key_policy_t g_builtin_keys_policy[] = {
+    {.key_id = TFM_BUILTIN_KEY_ID_HUK, .per_user_policy = 0, .usage = PSA_KEY_USAGE_DERIVE},
```

Comment:
This should be between #ifdef CONFIG_HW_UNIQUE_KEY, to be consistent with the rest.

Reply:
Agree